src/ucp/api/ucp.h

```diff
@@ -1614,6 +1614,12 @@ void ucp_listener_destroy(ucp_listener_h listener);
  * @param [out] ep_p        A handle to the created endpoint.
  *
  * @return Error code as defined by @ref ucs_status_t
+ *
+ * @note An endpoint destined to the same @a worker it's created on
+ * (@a params.address belongs to @a worker) will be connected to itself by default.
+ * This can be changed by passing the @ref UCP_EP_PARAMS_FLAGS_NO_LOOPBACK flag
+ * in @params.flags. It means the endpoint will become connected to *next*
+ * endpoint on the same @a worker which is created the same way.
  */
```

Comment:
@note By default, ucp_listener_destroy will connect an endpoint to itself if the endpoint is destined to the same @a worker on which it was created, i.e. @a params.address belongs to @a worker. This behavior can be changed by passing the @ref UCP_EP_PARAMS_FLAGS_NO_LOOPBACK flag in @params.flags. In that case, the endpoint will be connected to the *next* endpoint created in the same way on the same @a worker.